src/geometry/manifold/times_cont_mdiff_map.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Nicolò Cavalleri.
+-/
+
+import geometry.manifold.times_cont_mdiff
+import topology.continuous_map
+
+/-!
+# Smooth bundled map
+
+In this file we define the type `times_cont_mdiff_map` of `n` times continuously differentiable
+bundled maps.
+-/
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+{E : Type*} [normed_group E] [normed_space 𝕜 E]
+{E' : Type*} [normed_group E'] [normed_space 𝕜 E']
+{H : Type*} [topological_space H]
+{H' : Type*} [topological_space H']
+(I : model_with_corners 𝕜 E H) (I' : model_with_corners 𝕜 E' H')
+(M : Type*) [topological_space M] [charted_space H M] [smooth_manifold_with_corners I M]
+(M' : Type*) [topological_space M'] [charted_space H' M'] [smooth_manifold_with_corners I' M']
+{E'' : Type*} [normed_group E''] [normed_space 𝕜 E'']
+{H'' : Type*} [topological_space H'']
+{I'' : model_with_corners 𝕜 E'' H''}
+{M'' : Type*} [topological_space M''] [charted_space H'' M''] [smooth_manifold_with_corners I'' M'']
+(n : with_top ℕ)
+
+/-- Bundled `n` times continuously differentiable maps. -/
+@[protect_proj]
+structure times_cont_mdiff_map :=
+(to_fun                   : M → M')
+(times_cont_mdiff_to_fun  : times_cont_mdiff I I' n to_fun)
+
+/-- Bundled smooth maps. -/
+@[reducible] def smooth_map := times_cont_mdiff_map I I' M M' ⊤
+
+notation `C[` n `](` I `, ` M `; ` I' `, ` M' `)` := times_cont_mdiff_map I I' M M' n
+notation `C[` n `](` I `, ` M `; ` k `)` := times_cont_mdiff_map I (model_with_corners_self k k) M k n
+notation `C[` n `](` I `, ` M `)` := times_cont_mdiff_map I (model_with_corners_self ℝ ℝ) M ℝ n
```

Comment:
To me, `C^n(M)` is usually the space of smooth self-maps from `M` to itself, not the space of maps from `M` to the reals, so I would remove this one.

Reply:
I believe the notation of `C^n(M)` for real smooth maps is very widely used (and I never saw anybody using it differently). It is standard notation in Wikipedia, Math Stack Exchange and on all geometry textbooks I saw. However sure, let's remove it, it is not essential.